Source/core/Frame.h

```diff
@@ -582,15 +657,15 @@ namespace Core {
         }
 
         template <typename TYPENAME>
-        inline uint16_t GetNumber(const uint16_t offset, TYPENAME& value, const TemplateIntToType<false>&) const
+        inline SIZE_CONTEXT GetNumber(const SIZE_CONTEXT offset, TYPENAME& value, const TemplateIntToType<false>&) const
         {
             TYPENAME result;
 
             if ((offset + sizeof(TYPENAME)) > _size) {
                 ::memset(&result, 0, sizeof(result));
-            } else
+            }
+            else if (BIG_ENDIAN_ORDERING == true) {
 #ifdef LITTLE_ENDIAN_PLATFORM
-            {
```

Comment:
Didn't check very thoroughly but wouldn't it be better to put the endian handling in some helper method? Code is now duplicated it seems and an issue would need to be fixed in two places

Reply:
That was quicker to just copy/paste ;-) But created endian handling in separate methods,